zip-seanbowe-esktransfer.rst

```diff
@@ -0,0 +1,65 @@
+
+Naively, the recipient cannot know which ``(pk_d, d)`` was used to compute the shared secret, but the sender is asked to include the ``d`` within the Note plaintext to reconstruct the Note. However, if the recipient has many known addresses, an attacker could use a different payment address to perform secret exchange and, by observing the behavior of the recipient, link the two diversified addresses together.
+
+In order to prevent this attack, the protocol currently forces the sender to prove knowledge of the discrete logarithm of ``epk`` with respect to the ``g_d`` included within the Note commitment. This ``g_d`` is determined by ``d`` and recomputed during Note decryption, and so the recipient will either be unable to decrypt the note or the sender will be unable to perform the attack.
+
+However, this check occurs as part of the zero-knowledge proof statement and so relies on the knowledge soundness of the underlying zk-SNARK in Sapling. It would be preferable to force the sender to transfer ``esk`` within the Note plaintext so that, during note decryption, the recipient can ensure that it corresponds with ``epk`` and ``d`` correctly and ignore the payment as invalid otherwise. This allows a line of defense in the case that knowledge soundness of the zk-SNARK does not hold.
+
+Specification
+=============
+
+The transaction format is modified (in coordination with other proposals) so that the Sapling note plaintext must contain an additional 32-byte ``esk`` field.
+
+Sapling note decryption is defined to require that the ``epk = [esk] GH(d)`` for all valid Note plaintexts.
+
+Rationale
+=========
+
+The attack that this prevents is an interactive attack that requires an adversary to be able to break critical soundness properties of the zk-SNARKs underlying Sapling. It is potentially valid to assume that this cannot occur, due other damaging effects on the system such as undetectable counterfeiting. However, we have attempted to avoid any instance in the protocol where privacy (even against interactive attacks) depended on strong cryptographic assumptions. Acting differently here would be confusing for users that have previously been told that "privacy does not depend on zk-SNARK soundness" or similar claims.
+
+It is possible for us to infringe on the length of the ``memo`` field and ask the sender to provide ``esk`` within the existing Note plaintext without modifying the transaction format, but this would harm users that have come to expect a 512-byte memo field to be available to them. Changes to the memo field length should be considered in a broader context than changes made for cryptographic purposes.
+
+It is possible to transmit a signature of knowledge of a correct ``esk`` rather than ``esk`` itself, but this appears to be an unnecessary complication and is likely slower than just supplying ``esk``.
```

Comment:
This will need some more in-depth analysis of whether the assumption made of the encryption and key exchange needs to change. That doesn't block publication as a draft.